### PR TITLE
New module: ngsderive

### PR DESCRIPTION
**Pull request by claymcleod**
New module: **ngsderive**

 - [x] There is example tool output for tools in the https://github.com/ewels/MultiQC_TestData repository
     - Link to PR: https://github.com/ewels/MultiQC_TestData/pull/137/files
 - [x] Code is tested and works locally (including with `--lint` flag)
 - [x] `CHANGELOG.md` is updated
 - [x] `README.md` is updated
 - [x] `docs/README.md` is updated with link to below
 - [x] `docs/modulename.md` is created
 - [x] Everything that can be represented with a plot instead of a table is a plot
 - [x] Report sections have a description and help text (with `self.add_section`)
 - [x] There aren't any huge tables with > 6 columns (explain reasoning if so)
     - I did add a few columns to the summary table, but that is mainly because the output of `ngsderive` is categorical in nature.
 - [x] Each table column has a different colour scale to its neighbour, which relates to the data (eg. if high numbers are bad, they're red)
 - [x] Module does not do any significant computational work

### CODE REVIEW

**Comment by claymcleod** [comment]
@ewels, to my eye, the failure on the CI looks to be something not related to this PR. Do you agree with that assessment?

**Comment by claymcleod** [comment]
@ewels, there we go :). Let me know if there is anything more I can do. I’d appreciate a quick merge, as we are wanting to view these results in production. We could just pip install from master for the time being.

**Comment by claymcleod** [comment]
Hey @ewels, just wondering what sort of timeframe I should expect for a review + merge. Thanks!

**Comment by ewels** [comment]
Hi @claymcleod,

Getting to it as fast as I can, sorry for the lack of updates lately.

Phil

**Comment by ewels** [comment]
Hi @claymcleod,

You PR is next on my list and I'm just looking over it now. I've resolved the merge conflicts above, but when I try to run the code on the [test data](https://github.com/ewels/MultiQC_TestData/tree/master/data/modules/ngsderive) I get the following error:

```python
Module ngsderive raised an exception: Traceback (most recent call last):
  File "/Users/philewels/GitHub/MultiQC/multiqc/multiqc.py", line 546, in run
    output = mod()
  File "/Users/philewels/GitHub/MultiQC/multiqc/modules/ngsderive/ngsderive.py", line 53, in __init__
    EXPECTED_HEADER_COUNT_STRANDEDNESS
  File "/Users/philewels/GitHub/MultiQC/multiqc/modules/ngsderive/ngsderive.py", line 129, in parse
    "being malformed or an uncommon delimiter being detected.".format(sample_name=sample_name, subcommand=subcommand))
RuntimeError: Could not find results for {sample_name} in ngsderive {subcommand} report! This may be related to the file being malformed or an uncommon delimiter being detected.
```

Is this due to the test data being outdated? Or am I missing something here?

Cheers,

Phil

**Comment by ewels** [comment]
Yup, getting basically the same error on the linting: https://github.com/ewels/MultiQC/pull/1104/checks?check_run_id=605520637#step:8:11

**Comment by claymcleod** [comment]
Hey @ewels, let me take a look at it later today.

**Comment by ewels** [comment]
ok, let me know when the PR is ready for review 👍

**Comment by claymcleod** [comment]
@ewels I believe this should be ready for review again. @a-frantz was kind enough to pick up the development, so let us know if any questions arise.

File: multiqc/modules/ngsderive/ngsderive.py [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    ngsderive module class, parses stderr logs.
+    """
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="ngsderive",
+            anchor="ngsderive",
+            href="https://github.com/claymcleod/ngsderive",
```
Comment (claymcleod) [inline multiqc/modules/ngsderive/ngsderive.py:38]:
```suggestion
            href="https://github.com/stjudecloud/ngsderive",
```

@a-frantz

File: multiqc/modules/ngsderive/ngsderive.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+            info="is a forensic analysis tool useful in backwards computing information from "
+            + "next-generation sequencing data. "
+            + "Notably, results are provided as a 'best guess' — the tool does "
+            + "not claim 100%% accuracy and results should be considered with that understanding. "
+            + "Please see the <a href='https://github.com/claymcleod/ngsderive/#ngsderive'>documentation</a> "
```
Comment (claymcleod) [inline multiqc/modules/ngsderive/ngsderive.py:43]: @a-frantz same here

File: CHANGELOG.md [hunk 3]
```diff
@@ -87,6 +87,10 @@ Whilst it may be possible to continue using MultiQC with Python 2 for a short ti
 pinning dependencies, MultiQC compatibility for Python 2 will now slowly drift and start
 to break. If you haven't already, **you need to switch to Python 3 now**.
 
+#### New Modules:
+* [**ngsderive**](https://github.com/claymcleod/ngsderive)
```
Comment (claymcleod) [inline CHANGELOG.md:91]:
```suggestion
* [**ngsderive**](https://github.com/stjudecloud/ngsderive)
```

File: docs/modules/ngsderive.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+Name: ngsderive
+URL: https://github.com/claymcleod/ngsderive
```
Comment (claymcleod) [inline docs/modules/ngsderive.md:3]:
```suggestion
URL: https://github.com/stjudecloud/ngsderive
```

File: docs/modules/ngsderive.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+Description: >
+    ngsderive is a forensic analysis tool useful in backwards computing information from next-generation sequencing data.
+
+---
+
+`ngsderive` is a forensic analysis tool useful in backwards computing information from next-generation sequencing data. Notably, results are provided as a 'best guess' — the tool does not claim 100% accuracy and results should be considered with that understanding. Please see the <a href='https://github.com/claymcleod/ngsderive/#ngsderive'>documentation</a> for more information.
```
Comment (claymcleod) [inline docs/modules/ngsderive.md:9]:
```suggestion
`ngsderive` is a forensic analysis tool useful in backwards computing information from next-generation sequencing data. Notably, results are provided as a 'best guess' — the tool does not claim 100% accuracy and results should be considered with that understanding. Please see the <a href='stjudecloud.github.io/ngsderive'>documentation</a> for more information.
```

File: multiqc/modules/ngsderive/ngsderive.py [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+            "tt_percentages": True,
+            "ylab_format": "{value}%",
+            "cpswitch": False,
+        }
+
+        self.add_section(
+            name="Strandedness",
+            anchor="ngsderive-strandedness",
+            description="Predicted strandedness provided by ngsderive. "
+            + 'For more information, please see <a href="https://github.com/claymcleod/ngsderive/#strandedness-inference">'
```
Comment (claymcleod) [inline multiqc/modules/ngsderive/ngsderive.py:192]:
```suggestion
            + 'For more information, please see <a href="https://github.com/stjudecloud/ngsderive/#strandedness-inference">'
```

File: multiqc/modules/ngsderive/ngsderive.py [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+            "square": False,
+            "legend": False,
+            "datalabels": False,
+        }
+
+        self.add_section(
+            name="Instrument",
+            anchor="ngsderive-instrument",
+            description="Predicted instrument provided by ngsderive. "
+            + 'For more information, please see <a href="https://github.com/claymcleod/ngsderive/#illumina-machine-type">'
```
Comment (claymcleod) [inline multiqc/modules/ngsderive/ngsderive.py:272]:
```suggestion
            + 'For more information, please see <a href="https://stjudecloud.github.io/ngsderive/subcommands/instrument/">'
```

File: multiqc/modules/ngsderive/ngsderive.py [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+            "square": False,
+            "legend": False,
+            "datalabels": False,
+        }
+
+        self.add_section(
+            name="Read length",
+            anchor="ngsderive-readlen",
+            description="Predicted read length provided by ngsderive. "
+            + 'For more information, please see <a href="https://github.com/claymcleod/ngsderive/#read-length-calculation">'
```
Comment (claymcleod) [inline multiqc/modules/ngsderive/ngsderive.py:343]:
```suggestion
            + 'For more information, please see <a href="https://stjudecloud.github.io/ngsderive/subcommands/readlen/">'
```

**Comment by ewels** [comment]
Hi all,

Coming back to this again now, let's get this merged! 😁  Sorry it's taken me so long.

You now had merge conflicts and a lot of updates to pull in, so I did the rebase for you and sorted that out. I also started reviewing the code and making a few small tweaks and fixes.

I don't have permissions to push to your fork, so I created #1360 to continue this PR and review. Please go there for continued discussion 👍🏻 

Phil

**Pull request closed without merging**